### PR TITLE
PREAPPS-2454: 2FA | One time codes

### PR DESCRIPTION
**Pull request by panditsavitags**

### CODE REVIEW

File: src/schema/schema.graphql [hunk 1]
```diff
@@ -290,6 +290,10 @@ type ScratchCodeType {
 	_content: String
 }
 
+type ScratchCodes {
+	scratchCodes: ScratchCode
```
Comment (silentsakky) [inline src/schema/schema.graphql]: shouldn't this be an array? [ScratchCode]

File: src/batch-client/index.ts [hunk 2]
```diff
@@ -614,6 +626,18 @@ export class ZimbraBatchClient {
 			body: denormalize(GetRightsRequest)(options)
 		}).then(normalize(AccountRights));
 
+	public getScratchCodes = (username: String) =>
+		this.jsonRequest({
+			name: 'GetScratchCodes',
+			namespace: Namespace.Account,
+			body: {
```
Comment (SantoshKShetty) [inline src/batch-client/index.ts]: Is this payload required ? API doesn't say so...

Also, I doubt `namespace` is required here.

File: src/batch-client/index.ts [hunk 3]
```diff
@@ -489,6 +489,18 @@ export class ZimbraBatchClient {
 			}
 		}).then(res => normalize(FreeBusy)(res.usr));
 
+	public generateScratchCodes = (username: String) =>
+		this.jsonRequest({
+			name: 'GenerateScratchCodes',
+			namespace: Namespace.Account,
+			body: {
```
Comment (SantoshKShetty) [inline src/batch-client/index.ts]: Same as https://github.com/Zimbra/zm-api-js-client/pull/205/files#r292307415

File: src/batch-client/index.ts (hunk 2 above)
Comment (panditsavitags) [reply src/batch-client/index.ts]: yes, I tried this api without this payload, it won't work so i checked the api call that is being made by legacy which includes this payload. So without the account name this api does not work as expected

Comment (panditsavitags) [reply src/batch-client/index.ts]: the default namespace is mail, and this call is related to account so i added the namespace

Comment (panditsavitags) [reply src/batch-client/index.ts]: https://files.zimbra.com/docs/soap_api/8.8.8/api-reference/index.html , the namesapace mentioned is zimbraAccount

File: src/batch-client/index.ts (hunk 3 above)
Comment (silentsakky) [reply src/batch-client/index.ts]: username is mandatory parameter as per soap api

File: src/schema/schema.graphql (hunk 1 above)
Comment (pl12133) [reply src/schema/schema.graphql]: It looks like the API doesn't return an Array, https://files.zimbra.com/docs/soap_api/8.8.8/api-reference/zimbraAccount/GenerateScratchCodes.html

Comment (silentsakky) [reply src/schema/schema.graphql]: Yeah, weird implementation of apis :)

**Pull request merged**